lib/asset_cloud/base.rb

```diff
@@ -93,7 +91,7 @@ def initialize(root, url = '/')
     end
 
     def url_for(key, options={})
-      File.join(@url, URI.encode(key))
+      File.join(@url, Addressable::URI.encode_component(key, Addressable::URI::CharacterClasses::PATH))
```

Comment:
I couldn't find the docs at https://rubydoc.info/gems/addressable/Addressable/URI but still `Addressable::URI.encode_component` is properly documented in the repo, I don't know what's up with this.
Regarding the call, it seems that's how it's done in the gem itself, still kind of annoying there's not a method to do this directly.
```ruby
        :path => self.encode_component(uri_object.path,
          Addressable::URI::CharacterClasses::PATH),
```
https://github.com/sporkmonger/addressable/blob/addressable-2.7.0/lib/addressable/uri.rb#L613-L614